src/Glpi/Http/Kernel.php

```diff
@@ -0,0 +1,178 @@
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Http;
+
+use Symfony\Component\HttpKernel\Kernel as BaseKernel;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
+use Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait;
+
+class Kernel extends BaseKernel
+{
+    use MicroKernelTrait;
+
+    public function getProjectDir(): string
+    {
+        return \dirname(__DIR__, 3);
+    }
+
+    public function getCacheDir(): string
+    {
+        return $this->getProjectDir() . '/files/_cache/symfony';
+    }
+
+    public function getLogDir(): string
+    {
+        return $this->getProjectDir() . '/files/_log/';
+    }
+
+    public function registerBundles(): iterable
+    {
+        return [
+            new FrameworkBundle(),
+        ];
+    }
+
+    protected function configureContainer(ContainerConfigurator $container): void
+    {
+        $container
+            ->services()
+                ->load('Glpi\Controller\\', $this->getProjectDir() . '/src/Glpi/Controller')
+                    ->autowire()
+                    ->autoconfigure()
+        ;
+    }
+
+    protected function configureRoutes(RoutingConfigurator $routes): void
+    {
+        $routes->import($this->getProjectDir() . '/src/Glpi/Controller', 'attribute');
+    }
+
+    public function handle(Request $request, int $type = HttpKernelInterface::MAIN_REQUEST, bool $catch = true): Response
+    {
+        $this->runLegacyRouter($request);
+
+        return parent::handle($request, $type, $catch); // TODO: Change the autogenerated stub
+    }
+
+    private function runLegacyRouter(Request $request): void
+    {
+        if (defined('GLPI_ROOT')) {
+            // This var is defined in {root_dir}/index.php , the legacy front controller
+            return;
+        }
```

Comment:
I think it is not necessary. The `{root_dir}/index.php` will never include the `{root_dir}/public/index.php` file.